kong.conf.default

```diff
@@ -447,6 +447,22 @@
                                  # See the `lua_ssl_trusted_certificate`
                                  # setting to specify a certificate authority.
 
+#pg_max_concurrent_queries = 0   # Sets the maximum number of concurrent queries
+                                 # that can be executing at any given time. This
+                                 # limit is enforced per worker process; the
+                                 # total number of concurrent queries for this node
+                                 # will be will be `pg_max_concurrent_queries * nginx_worker_processes`.
+                                 # The default value of 0 removes this
+                                 # concurrency limitation.
+
+#pg_semaphore_timeout = 60       # Defines the timeout, in seconds, after which
+                                 # Postgres query semaphore resource acquisition
+                                 # attempts will fail. Such failures will
+                                 # generally result in the associated proxy or
+                                 # Admin API request failing with an HTTP 500.
```

Comment:
This makes me think that a desired user behavior _could be_ for Kong to catch such acquisition failure errors and exit with HTTP 503. Speculating here but at least such a behavior would be predictable for administrators and more explicit for clients than HTTP 500 (and more correct).

Reply:
(Yes, I did mean for the PostgreSQL connector to catch those failures and bluntly do something like: `if in_proxy then return ngx.exit(503)`)

Reply:
This feels like a reeeeally big hammer.

Do we even have such a `in_proxy` context/function? Also, why wouldn't we want to apply the same logic to failed acquisitions on the Admin API?

Reply:
Yes to the Admin API. No to the CLI or in-tests queries. I wonder how such "expected unexpected behavior" would be perceived by users, and if this would get any adoption at all in its current state.

Reply:
> if this would get any adoption at all in its current state.

Is the implication here that no one would use this feature at all unless semaphore acquisition failures returned a 503 instead of a 500? That seems like a really big stretch (apologies if I'm misreading).

I can make the change if it's a hard blocker for you. It seems like a big hammer and it's not the route I would take, but I'll defer to you here. Let me know.

Reply:
No, I don't have string feelings about it. Also got carried away a bit: even in its current state, we consistently return an error and the HTTP 500 will be predictable (arguably not an ideal status code but again, no strong feelings here), so I got unnecessarily worried.